pkg/odo/cli/url/list.go

```diff
@@ -53,17 +53,21 @@ func (o *URLListOptions) Run() (err error) {
 		return err
 	}
 
-	if len(urls.Items) == 0 {
-		return fmt.Errorf("no URLs found for component %v in application %v", o.Component(), o.Application)
+	if o.OutputFlag == "json" {
+		out, err := json.Marshal(urls)
+		if err != nil {
+			return err
+		}
+		fmt.Println(string(out))
```

Comment:
for me it returned, i think we should return an empty json in this case?
```
$ ./odo url list  -o json
 ✗  No component is set
```

Reply:
@syamgk , you are doing it from outside the context, this error is different, not related to `odo url list`

Reply:
Instead, try this `$ odo url list --context nodejs-ex/ -o json`

Reply:
yep